README.md

```diff
@@ -0,0 +1,203 @@
+
+The `build.sh` script will download and build a subset of JVips dependencies from source in order to maximize optimizations. However, it is recommended to install all dependencies on the system first, as documented in the sections below.
+
+Additionally, `build.sh` accepts the following options:
+- `--with-w64`, `--without-w64`: enable/disable Windows 64 build (default: disable)
+- `--with-linux`, `--without-linux`: enable/disable Linux build (default: enable)
+- `--with-macos`, `--without-macos`: enable/disable Linux build (default: disable)
+- `--skip-test`: disable unit tests (default: enable)
+- `--run-benchmark`: launch benchmark suite (default: disable)
+- `--dist`: build a `.tar.gz` archive containing all the build artifacts (default: disable)
+- `--minimal`: build 'minimal' Maven profile so JVips dependencies aren't embedded in the .jar file (default: all)
+- `--debug`: enable debugging in JVips and its dependencies (default: release mode)
+- `--jobs N`: use N jobs to build (default: 8)
+
+Run the build with:
+```
+$ ./build.sh [options]
+```
+
+Clean the project with:
+```
+$ ./clean.sh
+```
+
+### 🐧 Linux
+
+Install libvips development packages with your favorite package manager.
+
+For Ubuntu and Debian:
+```
+$ sudo apt-get install libvips-dev
+$ ./build.sh --without-w64 --with-linux --without-macos
+```
+
+For Fedora and CentOS:
+```
+$ sudo yum install vips-devel
+$ ./build.sh --without-w64 --with-linux --without-macos
+```
+
+### 🏁 Windows
+
+Windows builds are cross-compiled from Linux using MingW64. WSL and Docker are both supported.
+
+To build with Docker:
+```
+$ docker build --build-arg UID=$(id -u) --build-arg GID=$(id -g) -f Dockerfile -t builder .
+$ docker run --rm -v $(pwd):/app -w /app builder bash -ex build.sh --with-w64 --without-linux --without-macos
+```
+
+To build with WSL:
+```
+$ ./setup-for-ubuntu-wsl-w64-target.sh
+$ ./build.sh --with-w64 --without-linux --without-macos
+```
+
+### 🍎 macOS
+
+As macOS is mostly a development environment, JVips doesn't provide tools to build dependencies from sources and will always used system-wide libraries. The Homebrew packages installs all the required dependencies and headers to build JVips as is:
+```
+$ brew install vips
+$ ./build.sh --without-w64 --without-linux --with-macos
+```
+
+# Maintenance
+
+## Dependency upgrades
+
+Libraries are downloaded from hard-coded URL in `lib/build.sh`. Libraries can be upgraded by changing the version number in `lib/variable.sh`.
+
+## Benchmark
+
+To run a benchmark:
+1. `./build.sh --run-benchmark`
+
+On an Ubuntu 18.04 VM running with a quad-core Xeon E3-1271 v3 @ 3.6GHz, we apply the following operations:
+- open an 1920x1080 jpg image
+- resize to 100x100 dimension
+- crop a 50x30 rectangle on the top left corner
+- pad to a 50x80 image
+- write the output image into a buffer
+- release resources
```

Comment:
- resize to 512x512 dimension
- crop a 128x128 rectangle on the top left corner
- pad to a 256x256 image
- write the output image into a buffer
- release resources